admin/app/dfp/OrderAgent.scala

```diff
@@ -0,0 +1,20 @@
+package dfp
+
+import common.AkkaAgent
+import common.dfp.GuOrder
+
+import scala.concurrent.Future
+
+object OrderAgent {
+
+  private lazy val cache = AkkaAgent(Seq.empty[GuOrder])
+
+  def refresh(): Future[Seq[GuOrder]] = {
+    cache alterOff { oldData =>
+      val freshData = DfpApi.getAllOrders
+      if (freshData.nonEmpty) freshData else oldData
+    }
+  }
```

Comment:
I wonder if this refresh pattern can be encapsulated for different types, as it's repeated elsewhere.